tensorboard/backend/application.py

```diff
@@ -344,13 +373,17 @@ class are WSGI applications.
     parsed_url = urlparse.urlparse(request.path)
     clean_path = _clean_path(parsed_url.path, self._path_prefix)
 
+    @functools.wraps(start_response)
+    def new_start_response(status, headers):
```

Comment:
Was there a reason not to do this in `http_util.Respond()`?  That's where the other automatically populated headers are set: https://github.com/tensorflow/tensorboard/blob/63ab35a5a8e90a1e94d13b349a7ea56db8c748f4/tensorboard/backend/http_util.py#L148-L158

Reply:
Stephan and I talked about this briefly and thought that it was
reasonable to do in middleware. IMHO, we should consider moving most of
`Respond`’s functionality into middleware—our plugin API centers around
WSGI apps, which aren’t tied to Werkzeug or its `Request.application`
wrapper at all, so it’s strange to have this additional integration
point. What if plugin developers prefer Flask?

The alternative would be to make `http_util` a public module or move
`Request` into `plugin_util`.

Reply:
That's a fair point, but I guess it seems like then we should have a TODO/issue open to move that logic into the middleware, so there's a rationale available for why otherwise similar code is going in two different places.